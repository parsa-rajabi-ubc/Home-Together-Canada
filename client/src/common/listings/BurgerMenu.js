/**
 * @Author:     Parsa Rajabi
 * @Created:    2021.1.25
 *
 * @Description: Reusable Menu component
 *
 */
import React from 'react';
import PropTypes from "prop-types";
import Menu from 'react-burger-menu/lib/menus/slide';


const BurgerMenu = (props) => {
    const {isOpen, onClose, content, disableExistOnClickAway} = props;

    const styles = {
        bmOverlay: {
            background: 'transparent'
        }
    };

    return (
        //https://github.com/negomi/react-burger-menu
        <Menu
            styles={ styles }
            // disableOverlayClick = {disableExistOnClickAway}
            width={"30%"}
            disableAutoFocus
            itemListElement="div"
            customBurgerIcon={false}
            customCrossIcon={false}
            isOpen={isOpen} onClose={onClose}
            menuClassName={"rounded-xl"}
            itemListClassName={"bg-gray-700 overflow-auto"}
        >
            {content}
        </Menu>
    )
}

BurgerMenu.propTypes = {
    isOpen: PropTypes.bool,
    disableExistOnClickAway: PropTypes.bool,
    onClose: PropTypes.func,
    content: PropTypes.element
};

export default BurgerMenu;