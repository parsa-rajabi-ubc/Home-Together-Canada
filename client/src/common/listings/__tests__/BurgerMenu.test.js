/**
 * @Author:     Parsa Rajabi
 * @Created:    2021.1.25
 *
 * @Description: Test for BurgerMenu
 *
 */

import React from 'react';
import renderer from  'react-test-renderer'
import BurgerMenu from "../BurgerMenu";


describe('BurgerMenu', () => {
    describe('Snapshot test', () => {
        it("should render correctly regardless of properties", () => {
            // given
            const props = {
                isOpen: true,
                onClose: jest.fn(),
                content: <h1>Hello World</h1>
            };
            //when
            const component = renderer.create(<BurgerMenu {...props}/>).toJSON();
            //then
            expect(component).toMatchSnapshot();
        });
    })
})