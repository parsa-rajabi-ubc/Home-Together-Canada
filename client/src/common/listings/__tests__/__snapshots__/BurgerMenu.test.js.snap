// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BurgerMenu Snapshot test should render correctly regardless of properties 1`] = `
<div>
  <div
    className="bm-overlay"
    onClick={[Function]}
    style={
      Object {
        "MozTransform": "translate3d(100%, 0, 0)",
        "MsTransform": "translate3d(100%, 0, 0)",
        "OTransform": "translate3d(100%, 0, 0)",
        "WebkitTransform": "translate3d(100%, 0, 0)",
        "background": "transparent",
        "height": "100%",
        "opacity": 0,
        "position": "fixed",
        "transform": "translate3d(100%, 0, 0)",
        "transition": "opacity 0.3s, transform 0s 0.3s",
        "width": "100%",
        "zIndex": 1000,
      }
    }
  />
  <div
    aria-hidden={true}
    className="bm-menu-wrap"
    id=""
    style={
      Object {
        "MozTransform": "translate3d(-100%, 0, 0)",
        "MsTransform": "translate3d(-100%, 0, 0)",
        "OTransform": "translate3d(-100%, 0, 0)",
        "WebkitTransform": "translate3d(-100%, 0, 0)",
        "height": "100%",
        "position": "fixed",
        "right": "inherit",
        "transform": "translate3d(-100%, 0, 0)",
        "transition": "all 0.5s",
        "width": "30%",
        "zIndex": 1100,
      }
    }
  >
    <div
      className="bm-menu rounded-xl"
      style={
        Object {
          "boxSizing": "border-box",
          "height": "100%",
          "overflow": "auto",
        }
      }
    >
      <div
        className="bm-item-list bg-gray-700 overflow-auto"
        style={
          Object {
            "height": "100%",
          }
        }
      >
        <h1
          className="bm-item"
          style={
            Object {
              "display": "block",
            }
          }
          tabIndex={-1}
        >
          Hello World
        </h1>
      </div>
    </div>
  </div>
</div>
`;
